### ISSUE TITLE
Login fails if browser does not send `referer` 

### ISSUE THREAD

**Issue by paul90** [fedwiki/wiki-security-passportjs#14]
Over in the matrix room fortyfoxes [reported](https://matrix.to/#/!ORfrUEFeWFcHAMLFLr:matrix.org/$15097086024859977svrmt:matrix.org) an issue with logging into wiki when using Beaker.

Clicking on the padlock to signin, when using Beaker gives the following error:

```
TypeError: Parameter "url" must be a string, not undefined
    at Url.parse (url.js:81:11)
    at Object.urlParse [as parse] (url.js:75:5)
    at /usr/lib/node_modules/wiki/node_modules/wiki-security-passportjs/server/social.coffee:321:15
    at Layer.handle [as handle_request] (/usr/lib/node_modules/wiki/node_modules/express/lib/router/layer.js:95:5)
    at next (/usr/lib/node_modules/wiki/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/usr/lib/node_modules/wiki/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/usr/lib/node_modules/wiki/node_modules/express/lib/router/layer.js:95:5)
    at /usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:335:12)
    at next (/usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:275:10)
    at SessionStrategy.strategy.pass (/usr/lib/node_modules/wiki/node_modules/passport/lib/middleware/authenticate.js:325:9)
    at SessionStrategy.authenticate (/usr/lib/node_modules/wiki/node_modules/passport/lib/strategies/session.js:71:10)
    at attempt (/usr/lib/node_modules/wiki/node_modules/passport/lib/middleware/authenticate.js:348:16)
    at authenticate (/usr/lib/node_modules/wiki/node_modules/passport/lib/middleware/authenticate.js:349:7)
    at Layer.handle [as handle_request] (/usr/lib/node_modules/wiki/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:317:13)
    at /usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:335:12)
    at next (/usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:275:10)
    at initialize (/usr/lib/node_modules/wiki/node_modules/passport/lib/middleware/initialize.js:53:5)
    at Layer.handle [as handle_request] (/usr/lib/node_modules/wiki/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:317:13)
    at /usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:335:12)
    at next (/usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:275:10)
    at Layer.handle [as handle_request] (/usr/lib/node_modules/wiki/node_modules/express/lib/router/layer.js:91:12)
    at trim_prefix (/usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:317:13)
    at /usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:335:12)
    at next (/usr/lib/node_modules/wiki/node_modules/express/lib/router/index.js:275:10)
    at SendStream.error (/usr/lib/node_modules/wiki/node_modules/serve-static/index.js:121:7)
    at emitOne (events.js:96:13)
    at SendStream.emit (events.js:188:7)
    at SendStream.error (/usr/lib/node_modules/wiki/node_modules/send/index.js:270:17)
    at SendStream.onStatError (/usr/lib/node_modules/wiki/node_modules/send/index.js:421:12)
    at next (/usr/lib/node_modules/wiki/node_modules/send/index.js:736:16)
    at onstat (/usr/lib/node_modules/wiki/node_modules/send/index.js:725:14)
    at FSReqWrap.oncomplete (fs.js:123:15)
```

*As an aside, it is worth noting that the referer header is optional, so is not guaranteed to be present. So, switching to a different method of passing the wiki that is being logged into is good.*

**Comment by paul90** [comment]
Since this was raised there is some thought that in a future version of Beaker, when Apps are fully working, that Javascript will be blocked excerpt from Apps/dat sites.

Fixing this issue is still relevant, as referer is seen as a privacy concern and not passed by some.

**Comment by paul90** [comment]
Fixed by #15, published as + wiki-security-passportjs@0.1.4

**Comment by paul90** [comment]
Have pulled wiki-security-passportjs@0.1.4 as it does not behave well...

**Comment by paul90** [comment]
This was fixed in v0.2.0
